corehq/apps/es/users.py

```diff
@@ -74,8 +81,52 @@ def mapping(self):
         return get_adapter_mapping(self)
 
     @classmethod
-    def from_python(cls, doc):
-        return from_dict_with_possible_id(doc)
+    @classmethod
+    def from_multi(cls, user):
+        """
+        Takes in a user object or a user dict and applies required transfomation to make it suitable for ES.
+        The function is replica of ``transform_user_for_elasticsearch`` with added support for user objects.
+        In future all refernces to  ``transform_user_for_elasticsearch`` will be replaced by `from_python`
+
+        :param user: an instance of ``CouchUser`` or ``dict`` which is ``couch_user.to_json()``
+
+        :raises InvalidDictForAdapter: if ``dict`` passed is not valid user dict
+        :raises UnknownDocException: if object passes in not instance of ``CouchUser``
+        """
+        from corehq.apps.groups.dbaccessors import (
+            get_group_id_name_map_by_user,
+        )
+        from corehq.apps.users.models import CouchUser
+
+        def _verify_and_return_user_as_dict():
+            if isinstance(user, dict):
+                try:
+                    CouchUser.wrap_correctly(user)
+                    return deepcopy(user)
```

Comment:
> Can this be replaced with a shallow copy? 

Sure. Sounds like a good optimization.


> In the case of user_dict['user_data_es'].append(...) below, it could do something like

Just for my clarity, why do you think append should not be used here?
```
      user_dict['user_data_es'] = []
      if 'user_data' in user_dict:
          user_obj = CouchUser.wrap_correctly(user_dict)
          for key, value in user_obj.metadata.items():
              user_dict['user_data_es'].append({
                  'key': key,
                  'value': value,
              })
```
 We are actually creating a new array and appending values to it, would it matter if it is modified in place?
We are creating a new array in the dict, so append is updating the new list not the

Reply:
Oh, I missed this line: `user_dict['user_data_es'] = []`

Yes, should be safe to append.